### PR TITLE
Refactor to cleanup after selector report range fixes

### PR DESCRIPTION
**Pull request by romainmenke**
> Which issue, if any, is this issue related to?

See: https://github.com/stylelint/stylelint/issues/7904

> Is there anything in the PR that needs further explanation?

After all the fixes for the reported ranges of selector problems there were still a few opportunities to cleanup existing code, slightly improve test coverage, and a few very minor improvements.

I bundled these because they were so minor but I also don't mind splitting this up.

I don't think a changeset is needed because most changes aren't user facing and those that are, already have a relevant changeset.

### CODE REVIEW

File: lib/rules/selector-disallowed-list/__tests__/index.mjs [hunk 1]
```diff
@@ -203,7 +203,7 @@ testRule({
 		},
 		{
 			code: '.foo > /*comment*/.bar {}',
-			message: messages.rejected('.foo > .bar'),
+			message: messages.rejected('.foo > /*comment*/.bar'),
```
Comment (romainmenke) [inline lib/rules/selector-disallowed-list/__tests__/index.mjs:206]: This is user facing, but this is a minor improvement that follows on previous work to fix the reported ranges.

See: https://github.com/stylelint/stylelint/blob/971b61e85564975e9b2741f7c286ea248e3b243e/.changeset/sixty-kings-join.md?plain=1#L5

Comment (Mouvedia) [reply lib/rules/selector-disallowed-list/__tests__/index.mjs:206]: Remember to add `& [@romainmenke](https://github.com/romainmenke)`

File: lib/rules/selector-disallowed-list/index.cjs [hunk 2]
```diff
@@ -75,38 +75,38 @@ const rule = (primary, secondaryOptions) => {
...
+						messageArgs: [selectorStr],
 						node: ruleNode,
-						word,
+						index,
+						endIndex,
 					});
 				}
 			}
+
+			const selectors = getRuleSelector(ruleNode);
```
Comment (Mouvedia) [inline lib/rules/selector-disallowed-list/index.cjs:97]: nit

We have 3 cases of singular and one plural.
i.e. maybe rename `selector`

Comment (Mouvedia) [inline lib/rules/selector-disallowed-list/index.cjs:97]: nit

maybe rename to `selector`
it makes more sense when passing it to `parseSlector`

File: lib/rules/selector-type-case/index.mjs [hunk 3]
```diff
@@ -48,8 +49,7 @@ const rule = (primary, secondaryOptions) => {
 		}
 
 		root.walkRules((ruleNode) => {
-			let hasComments = ruleNode.raws.selector && ruleNode.raws.selector.raw;
-			const selector = hasComments ? hasComments : ruleNode.selector;
+			const selector = ruleNode.selector;
```
Comment (Mouvedia) [inline lib/rules/selector-type-case/index.mjs:52]: question

Is this on purpose?
i.e. didn't you want `const selector = getRuleSelector(ruleNode);` instead?

File: lib/rules/selector-disallowed-list/index.cjs (hunk 2 above)
Comment (romainmenke) [reply lib/rules/selector-disallowed-list/index.cjs:97]: Can't, `selector` is already used

File: lib/rules/selector-type-case/index.mjs (hunk 3 above)
Comment (romainmenke) [reply lib/rules/selector-type-case/index.mjs:52]: No, this is on purpose.

Here `ruleNode.selector` is only used for fast aborts.

File: lib/rules/selector-disallowed-list/__tests__/index.mjs (hunk 1 above)
Comment (romainmenke) [reply lib/rules/selector-disallowed-list/__tests__/index.mjs:206]: I don't really need the attribution unless there is a requirement to add these? ☺️

File: lib/rules/selector-class-pattern/index.mjs [hunk 4]
```diff
@@ -50,19 +50,17 @@ const rule = (primary, secondaryOptions) => {
...
-			if (selectors.some((s) => isKeyframeSelector(s))) {
+			if (ruleNode.selectors.some(isKeyframeSelector)) {
 				return;
 			}
 
 			// Only bother resolving selectors that have an interpolating &
-			if (shouldResolveNestedSelectors && hasInterpolatingAmpersand(selector)) {
-				for (const nestedSelector of resolveNestedSelector(selector, ruleNode)) {
+			if (shouldResolveNestedSelectors && hasInterpolatingAmpersand(ruleNode.selector)) {
+				for (const nestedSelector of resolveNestedSelector(getRuleSelector(ruleNode), ruleNode)) {
```
Comment (Mouvedia) [inline lib/rules/selector-class-pattern/index.mjs:63]: The changelog will have to be edited for this one too.

File: lib/rules/selector-type-case/index.mjs
Comment (Mouvedia) [inline lib/rules/selector-type-case/index.mjs:1]: great code reduction

**Review by Mouvedia — APPROVED** [review]

**Comment by romainmenke** [comment]
Thank you for reviewing this and for the feedback @Mouvedia 🙇

File: lib/rules/selector-disallowed-list/index.cjs (hunk 2 above)
Comment (Mouvedia) [reply lib/rules/selector-disallowed-list/index.cjs:97]: maybe `ruleSelector` was available
anyway, not important

**Pull request merged**